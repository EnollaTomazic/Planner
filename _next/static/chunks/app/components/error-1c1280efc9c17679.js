(self.webpackChunk_N_E=self.webpackChunk_N_E||[]).push([[872],{50960:(e,s,r)=>{"use strict";r.d(s,{A:()=>o,R:()=>c});var a=r(95155),n=r(12115),i=r(52619),t=r.n(i),l=r(24625);function c(e){let{error:s,reset:r,title:i,description:c,retryLabel:o="Try again",homeLabel:d="Go to dashboard",homeHref:h="/"}=e;return(0,n.useEffect)(()=>{console.error(s)},[s]),(0,a.jsx)(l.qK,{as:"section",role:"alert","aria-live":"assertive",className:"py-[var(--space-8)]",children:(0,a.jsxs)("div",{className:"flex max-w-prose flex-col gap-[var(--space-4)]",children:[(0,a.jsxs)("div",{className:"space-y-[var(--space-2)]",children:[(0,a.jsx)("h1",{className:"text-display-sm font-semibold text-foreground",children:i}),(0,a.jsx)("p",{className:"text-body text-muted-foreground",children:c})]}),(0,a.jsxs)("div",{className:"flex flex-wrap gap-[var(--space-3)]",children:[(0,a.jsx)(l.$n,{variant:"primary",onClick:r,children:o}),(0,a.jsx)(l.$n,{asChild:!0,variant:"ghost",children:(0,a.jsx)(t(),{href:h,children:d})})]})]})})}function o(e){return(0,a.jsx)(c,{...e,title:"Something went wrong",description:"This section hit an error, but the rest of Planner is still running. Try again or return home."})}},59729:(e,s,r)=>{"use strict";r.r(s),r.d(s,{default:()=>a.A});var a=r(50960)},94822:(e,s,r)=>{Promise.resolve().then(r.bind(r,59729))}},e=>{e.O(0,[6428,9295,2619,4625,8441,1255,7358],()=>e(e.s=94822)),_N_E=e.O()}]);