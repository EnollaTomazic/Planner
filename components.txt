1:"$Sreact.fragment"
2:I[41402,["678","static/chunks/678-3b8ca7f6c2bfafc6.js","619","static/chunks/619-f072ac750404f9da.js","625","static/chunks/625-de68f502e68d6418.js","177","static/chunks/app/layout-ff43e66cdbce5daf.js"],""]
3:I[28727,["678","static/chunks/678-3b8ca7f6c2bfafc6.js","619","static/chunks/619-f072ac750404f9da.js","625","static/chunks/625-de68f502e68d6418.js","177","static/chunks/app/layout-ff43e66cdbce5daf.js"],"default"]
4:I[98002,["678","static/chunks/678-3b8ca7f6c2bfafc6.js","619","static/chunks/619-f072ac750404f9da.js","625","static/chunks/625-de68f502e68d6418.js","177","static/chunks/app/layout-ff43e66cdbce5daf.js"],"default"]
5:I[9766,[],""]
6:I[98924,[],""]
7:I[41061,["678","static/chunks/678-3b8ca7f6c2bfafc6.js","619","static/chunks/619-f072ac750404f9da.js","689","static/chunks/689-e5d24021d927ff8f.js","736","static/chunks/736-4db4399c886d749f.js","634","static/chunks/634-bfd3c8aa4039d99b.js","625","static/chunks/625-de68f502e68d6418.js","373","static/chunks/373-f15cb54580f283eb.js","108","static/chunks/108-03d12cb04b17ced4.js","522","static/chunks/522-4c109c4a359a5202.js","879","static/chunks/879-fd6a10fb62d46999.js","423","static/chunks/app/components/page-73e48587b2bb0ffd.js"],"default"]
8:I[52619,["678","static/chunks/678-3b8ca7f6c2bfafc6.js","619","static/chunks/619-f072ac750404f9da.js","625","static/chunks/625-de68f502e68d6418.js","345","static/chunks/app/not-found-e4eee859b0a932fb.js"],""]
9:I[93462,["678","static/chunks/678-3b8ca7f6c2bfafc6.js","619","static/chunks/619-f072ac750404f9da.js","689","static/chunks/689-e5d24021d927ff8f.js","736","static/chunks/736-4db4399c886d749f.js","634","static/chunks/634-bfd3c8aa4039d99b.js","625","static/chunks/625-de68f502e68d6418.js","373","static/chunks/373-f15cb54580f283eb.js","108","static/chunks/108-03d12cb04b17ced4.js","522","static/chunks/522-4c109c4a359a5202.js","879","static/chunks/879-fd6a10fb62d46999.js","423","static/chunks/app/components/page-73e48587b2bb0ffd.js"],"default"]
f:I[57150,[],""]
:HL["/Planner/_next/static/css/eee3817ff2c519e0.css","style"]
:HL["/Planner/_next/static/css/cb150f8d5a7ebde5.css","style"]
:HL["/Planner/_next/static/css/9d79e8c8cefd0d15.css","style"]
:HL["/Planner/_next/static/css/4e282041685a4199.css","style"]
:HL["/Planner/_next/static/css/fac86c4fdf6106a7.css","style"]
0:{"P":null,"b":"Fmq1Vd77VM9lxh_JyjaIU","p":"/Planner","c":["","components"],"i":false,"f":[[["",{"children":["components",{"children":["__PAGE__",{}]}]},"$undefined","$undefined",true],["",["$","$1","c",{"children":[[["$","link","0",{"rel":"stylesheet","href":"/Planner/_next/static/css/eee3817ff2c519e0.css","precedence":"next","crossOrigin":"$undefined","nonce":"$undefined"}],["$","link","1",{"rel":"stylesheet","href":"/Planner/_next/static/css/cb150f8d5a7ebde5.css","precedence":"next","crossOrigin":"$undefined","nonce":"$undefined"}],["$","link","2",{"rel":"stylesheet","href":"/Planner/_next/static/css/9d79e8c8cefd0d15.css","precedence":"next","crossOrigin":"$undefined","nonce":"$undefined"}]],["$","html",null,{"lang":"en","className":"theme-lg","suppressHydrationWarning":true,"style":{"--asset-noise-url":"url('/Planner/noise.svg')","--asset-glitch-gif-url":"url('/Planner/glitch-gif.gif')"},"children":[["$","head",null,{"children":["$","$L2",null,{"id":"theme-bootstrap","strategy":"beforeInteractive","src":"/Planner/scripts/theme-bootstrap.js"}]}],["$","body",null,{"className":"min-h-screen bg-background text-foreground glitch-root","children":[["$","a",null,{"className":"fixed left-[var(--space-4)] top-[var(--space-4)] z-50 inline-flex items-center rounded-[var(--radius-lg)] bg-background px-[var(--space-4)] py-[var(--space-2)] text-ui font-medium text-foreground shadow-outline-subtle outline-none transition-all duration-[var(--dur-quick)] ease-[var(--ease-out)] opacity-0 -translate-y-full pointer-events-none focus-visible:translate-y-0 focus-visible:opacity-100 focus-visible:pointer-events-auto focus-visible:shadow-ring focus-visible:no-underline focus-visible:outline-none hover:shadow-ring focus-visible:active:translate-y-[var(--space-1)]","href":"#main-content","children":"Skip to main content"}],["$","$L3",null,{"children":[["$","div",null,{"aria-hidden":true,"className":"page-backdrop","children":["$","div",null,{"className":"page-shell","children":["$","div",null,{"className":"page-backdrop__layer"}]}]}],["$","$L4",null,{}],["$","div",null,{"className":"fixed bottom-[var(--space-4)] left-[var(--space-4)] z-50 pointer-events-none select-none","children":["$","svg",null,{"className":"w-[var(--space-8)] h-[var(--space-8)] text-accent motion-safe:animate-[cat-float_3s_ease-in-out_infinite]","role":"img","aria-label":"Cat companion","tabIndex":-1,"focusable":"false","viewBox":"0 0 24 24","fill":"currentColor","children":["$","path",null,{"d":"M12 5c.67 0 1.35.09 2 .26 1.78-2 5.03-2.84 6.42-2.26 1.4.58-.42 7-.42 7 .57 1.07 1 2.24 1 3.44C21 17.9 16.97 21 12 21s-9-3-9-7.56c0-1.25.5-2.4 1-3.44 0 0-1.89-6.42-.5-7 1.39-.58 4.72.23 6.5 2.23A9.04 9.04 0 0 1 12 5Z"}]}]}],["$","div",null,{"id":"main-content","tabIndex":-1,"className":"relative z-10","children":["$","$L5",null,{"parallelRouterKey":"children","error":"$undefined","errorStyles":"$undefined","errorScripts":"$undefined","template":["$","$L6",null,{}],"templateStyles":"$undefined","templateScripts":"$undefined","notFound":[["$","main",null,{"className":"flex min-h-screen flex-col items-center justify-center gap-4 p-6 text-center","aria-labelledby":"not-found-header","children":["$","$L7",null,{"className":"rounded-card r-card-lg px-4 py-4","header":{"id":"not-found-header","heading":"Page not found","icon":["$","svg",null,{"ref":"$undefined","xmlns":"http://www.w3.org/2000/svg","width":24,"height":24,"viewBox":"0 0 24 24","fill":"none","stroke":"currentColor","strokeWidth":2,"strokeLinecap":"round","strokeLinejoin":"round","className":"lucide lucide-circle-alert opacity-80","aria-hidden":"true","children":[["$","circle","1mglay",{"cx":"12","cy":"12","r":"10"}],["$","line","1pkeuh",{"x1":"12","x2":"12","y1":"8","y2":"12"}],["$","line","4dfq90",{"x1":"12","x2":"12.01","y1":"16","y2":"16"}],"$undefined"]}]},"hero":{"frame":false,"heading":"This page does not exist","actions":["$","$L8",null,{"href":"/","children":["$","$L9",null,{"className":"px-4","children":"Go home"}]}],"children":["$","p",null,{"className":"text-ui text-muted-foreground","children":"The page you are looking for does not exist."}]}}]}],[]],"forbidden":"$undefined","unauthorized":"$undefined"}]}]]}]]}]]}]]}],{"children":["components",["$","$1","c",{"children":[null,["$","$L5",null,{"parallelRouterKey":"children","error":"$undefined","errorStyles":"$undefined","errorScripts":"$undefined","template":["$","$L6",null,{}],"templateStyles":"$undefined","templateScripts":"$undefined","notFound":"$undefined","forbidden":"$undefined","unauthorized":"$undefined"}]]}],{"children":["__PAGE__",["$","$1","c",{"children":["$La",["$Lb","$Lc"],"$Ld"]}],{},null,false]},null,false]},null,false],"$Le",false]],"m":"$undefined","G":["$f",[]],"s":false,"S":true}
10:"$Sreact.suspense"
11:I[33173,["678","static/chunks/678-3b8ca7f6c2bfafc6.js","619","static/chunks/619-f072ac750404f9da.js","689","static/chunks/689-e5d24021d927ff8f.js","736","static/chunks/736-4db4399c886d749f.js","634","static/chunks/634-bfd3c8aa4039d99b.js","625","static/chunks/625-de68f502e68d6418.js","373","static/chunks/373-f15cb54580f283eb.js","108","static/chunks/108-03d12cb04b17ced4.js","522","static/chunks/522-4c109c4a359a5202.js","879","static/chunks/879-fd6a10fb62d46999.js","423","static/chunks/app/components/page-73e48587b2bb0ffd.js"],"default"]
12:I[26628,["678","static/chunks/678-3b8ca7f6c2bfafc6.js","619","static/chunks/619-f072ac750404f9da.js","689","static/chunks/689-e5d24021d927ff8f.js","736","static/chunks/736-4db4399c886d749f.js","634","static/chunks/634-bfd3c8aa4039d99b.js","625","static/chunks/625-de68f502e68d6418.js","373","static/chunks/373-f15cb54580f283eb.js","108","static/chunks/108-03d12cb04b17ced4.js","522","static/chunks/522-4c109c4a359a5202.js","879","static/chunks/879-fd6a10fb62d46999.js","423","static/chunks/app/components/page-73e48587b2bb0ffd.js"],"default"]
13:I[24431,[],"OutletBoundary"]
15:I[15278,[],"AsyncMetadataOutlet"]
17:I[24431,[],"ViewportBoundary"]
19:I[24431,[],"MetadataBoundary"]
a:["$","$10",null,{"fallback":["$","div",null,{"className":"flex justify-center p-6","children":["$","$L11",null,{}]}],"children":["$","$L12",null,{}]}]
b:["$","link","0",{"rel":"stylesheet","href":"/Planner/_next/static/css/4e282041685a4199.css","precedence":"next","crossOrigin":"$undefined","nonce":"$undefined"}]
c:["$","link","1",{"rel":"stylesheet","href":"/Planner/_next/static/css/fac86c4fdf6106a7.css","precedence":"next","crossOrigin":"$undefined","nonce":"$undefined"}]
d:["$","$L13",null,{"children":["$L14",["$","$L15",null,{"promise":"$@16"}]]}]
e:["$","$1","h",{"children":[null,[["$","$L17",null,{"children":"$L18"}],null],["$","$L19",null,{"children":["$","div",null,{"hidden":true,"children":["$","$10",null,{"fallback":null,"children":"$L1a"}]}]}]]}]
18:[["$","meta","0",{"charSet":"utf-8"}],["$","meta","1",{"name":"viewport","content":"width=device-width, initial-scale=1"}]]
14:null
16:{"metadata":[["$","title","0",{"children":"Components Â· Planner"}],["$","meta","1",{"name":"description","content":"Browse Planner UI building blocks and examples."}]],"error":null,"digest":"$undefined"}
1a:"$16:metadata"
