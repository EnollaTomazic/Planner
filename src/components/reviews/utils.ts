import type * as React from "react";

export type Result = "Win" | "Loss";

/** Parse "m:ss" or "mm:ss" into seconds. Returns null for invalid input. */
export function parseTime(mmss: string): number | null {
  const m = mmss.trim().match(/^(\d{1,2}):([0-5]\d)$/);
  if (!m) return null;
  return Number(m[1]) * 60 + Number(m[2]);
}

/** Convert seconds to "m:ss" with zero-padded seconds. */
export function formatSeconds(total: number): string {
  const minutes = Math.max(0, Math.floor(total / 60));
  const seconds = Math.max(0, total % 60);
  return `${String(minutes)}:${String(seconds).padStart(2, "0")}`;
}

export function onIconKey(
  e: React.KeyboardEvent,
  handler: () => void,
): void {
  if (e.key === " " || e.key === "Enter") {
    e.preventDefault();
    handler();
  }
}
