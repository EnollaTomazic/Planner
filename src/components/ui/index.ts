// Auto-generated by scripts/generate-ui-index.ts
// Do not edit directly.
export { default as AnimationToggle } from "./AnimationToggle";
export { default as Badge } from "./Badge";
export * from "./Badge";
export { default as Label } from "./Label";
export * from "./Label";
export { default as Select } from "./Select";
export * from "./Select";
export { default as Progress } from "./feedback/Progress";
export { default as Snackbar } from "./feedback/Snackbar";
export { default as Spinner } from "./feedback/Spinner";
export { default as Header } from "./layout/Header";
export * from "./layout/Header";
export { default as Hero2 } from "./layout/Hero2";
export * from "./layout/Hero2";
export { default as SectionCard } from "./layout/SectionCard";
export { default as Split } from "./layout/Split";
export { default as TabBar } from "./layout/TabBar";
export * from "./layout/TabBar";
export { default as TitleBar } from "./layout/TitleBar";
export { default as SideSelector } from "./league/SideSelector";
export * from "./league/SideSelector";
export { default as PillarBadge } from "./league/pillars/PillarBadge";
export { default as PillarSelector } from "./league/pillars/PillarSelector";
export { default as Button } from "./primitives/Button";
export * from "./primitives/Button";
export { default as Card } from "./primitives/Card";
export * from "./primitives/Card";
export { default as FieldShell } from "./primitives/FieldShell";
export * from "./primitives/FieldShell";
export * from "./primitives/GlitchSegmented";
export { default as IconButton } from "./primitives/IconButton";
export * from "./primitives/IconButton";
export { default as Input } from "./primitives/Input";
export * from "./primitives/Input";
export * from "./primitives/Neu";
export { default as SearchBar } from "./primitives/SearchBar";
export * from "./primitives/SearchBar";
export { default as Textarea } from "./primitives/Textarea";
export * from "./primitives/Textarea";
export { default as AnimatedSelect } from "./selects/AnimatedSelect";
export * from "./selects/AnimatedSelect";
export { default as ThemeToggle } from "./theme/ThemeToggle";
export { default as CheckCircle } from "./toggles/CheckCircle";
export * from "./toggles/NeonIcon";
export { default as Toggle } from "./toggles/toggle";
