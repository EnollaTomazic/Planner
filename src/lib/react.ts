import * as React from "react";

export const hasTextContent = (node: React.ReactNode): boolean => {
  if (node === null || node === undefined) return false;
  if (typeof node === "boolean") return false;
  if (typeof node === "string") return node.trim().length > 0;
  if (typeof node === "number") return true;
  if (Array.isArray(node)) return node.some((item) => hasTextContent(item));
  if (React.isValidElement<{ children?: React.ReactNode }>(node)) {
    return hasTextContent(node.props.children ?? null);
  }
  return false;
};

export function observeMediaQuery(
  query: string,
  onChange: (matches: boolean, mediaQuery: MediaQueryList) => void,
): (() => void) | undefined {
  if (typeof window === "undefined" || typeof window.matchMedia !== "function") {
    return undefined;
  }

  let mediaQuery: MediaQueryList;

  try {
    mediaQuery = window.matchMedia(query);
  } catch {
    return undefined;
  }

  const handleChange = (event: MediaQueryListEvent): void => {
    onChange(event.matches, mediaQuery);
  };

  const handleLegacyChange = (): void => {
    onChange(mediaQuery.matches, mediaQuery);
  };

  onChange(mediaQuery.matches, mediaQuery);

  if (typeof mediaQuery.addEventListener === "function") {
    mediaQuery.addEventListener("change", handleChange);
    return () => {
      mediaQuery.removeEventListener("change", handleChange);
    };
  }

  if (typeof mediaQuery.addListener === "function") {
    mediaQuery.addListener(handleLegacyChange);
    return () => {
      mediaQuery.removeListener(handleLegacyChange);
    };
  }

  return undefined;
}

export function useMatchMedia(
  query: string,
  getServerSnapshot?: () => boolean,
): boolean {
  const getMatches = React.useCallback(() => {
    if (typeof window === "undefined") {
      return typeof getServerSnapshot === "function" ? getServerSnapshot() : false;
    }

    if (typeof window.matchMedia !== "function") {
      return false;
    }

    try {
      return window.matchMedia(query).matches;
    } catch {
      return false;
    }
  }, [query, getServerSnapshot]);

  const [matches, setMatches] = React.useState<boolean>(getMatches);

  React.useEffect(() => {
    return observeMediaQuery(query, (nextMatches) => {
      setMatches(nextMatches);
    });
  }, [query]);

  return matches;
}
