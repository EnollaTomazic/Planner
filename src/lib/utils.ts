// src/lib/utils.ts
// Tiny helpers. Keep dependencies minimal and SSR-safe.

import { type ClassValue, clsx } from "clsx";
import { extendTailwindMerge } from "tailwind-merge";

const twMerge = extendTailwindMerge({
  extend: {
    theme: {
      text: ["label", "ui", "body", "title", "title-lg"],
    },
  },
});

const RAW_BASE_PATH = process.env.NEXT_PUBLIC_BASE_PATH ?? "";
const NORMALIZED_BASE =
  RAW_BASE_PATH && RAW_BASE_PATH !== "/"
    ? `/${RAW_BASE_PATH.replace(/^\/+|\/+$|\s+/g, "")}`
    : "";

const ABSOLUTE_URL_PATTERN = /^[a-zA-Z][a-zA-Z\d+.-]*:/;

function isAbsoluteUrl(path: string): boolean {
  return ABSOLUTE_URL_PATTERN.test(path) || path.startsWith("//");
}

// Default locale for consistent date/time formatting.
export const LOCALE = "en-US";

/** Combine class names using clsx and tailwind-merge. */
export function cn(...inputs: ClassValue[]): string {
  return twMerge(clsx(inputs));
}

/**
 * Prefix a path with the configured Next.js base path, if any.
 * Ensures consistent asset URLs for environments served from sub-paths.
 */
export function withBasePath(path: string): string {
  if (isAbsoluteUrl(path)) {
    return path;
  }
  const normalizedPath = path.startsWith("/") ? path : `/${path}`;

  if (!NORMALIZED_BASE) {
    return normalizedPath;
  }

  return `${NORMALIZED_BASE}${normalizedPath}`;
}

/** Remove the configured base path prefix from a pathname. */
export function withoutBasePath(path: string): string {
  if (isAbsoluteUrl(path)) {
    return path;
  }
  const normalizedPath = path.startsWith("/") ? path : `/${path}`;

  if (!NORMALIZED_BASE) {
    return normalizedPath;
  }

  if (
    normalizedPath === NORMALIZED_BASE ||
    normalizedPath === `${NORMALIZED_BASE}/`
  ) {
    return "/";
  }

  if (normalizedPath.startsWith(`${NORMALIZED_BASE}/`)) {
    const remainder = normalizedPath.slice(NORMALIZED_BASE.length);
    return remainder.length > 0 ? remainder : "/";
  }

  return normalizedPath;
}

/** Capitalize first letter (not Unicode-smart on purpose). */
export function capitalize(s: string): string {
  return s.length ? s[0].toUpperCase() + s.slice(1) : s;
}

/** Clamp number to a range. */
export function clamp(n: number, min: number, max: number): number {
  return Math.min(max, Math.max(min, n));
}

// Date helpers moved to ./date.ts

/** Create a URL-friendly slug from a string. */
export function slugify(s?: string): string {
  return (s ?? "")
    .toLowerCase()
    .trim()
    .replace(/\s+/g, "-")
    .replace(/[^a-z0-9-]/g, "")
    .slice(0, 64);
}

/** Escape mappings for sanitizeText */
const HTML_ESCAPE_MAP = {
  "&": "&amp;",
  "<": "&lt;",
  ">": "&gt;",
  "\u0022": "&quot;",
  "'": "&#39;",
} as const;

/**
 * Clone data using structuredClone with JSON fallback.
 * Returns undefined when cloning fails.
 */
export function safeClone<T>(value: T): T | undefined {
  if (typeof structuredClone === "function") {
    try {
      return structuredClone(value);
    } catch {
      // fall through to JSON
    }
  }
  try {
    return JSON.parse(JSON.stringify(value)) as T;
  } catch {
    return undefined;
  }
}

/**
 * sanitizeText â€” escape HTML-unsafe characters to prevent node injection.
 * Minimal on purpose; more heavy sanitizers can be added if needed.
 */
export function sanitizeText(input: string): string {
  return input.replace(
    /[&<>"']/g,
    (c) => HTML_ESCAPE_MAP[c as keyof typeof HTML_ESCAPE_MAP] ?? c,
  );
}
