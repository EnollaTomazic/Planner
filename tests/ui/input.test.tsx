import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import { describe, it, expect, afterEach } from 'vitest';
import { Input } from '@/components/ui';
import { slugify } from '@/lib/utils';

afterEach(cleanup);

describe('Input', () => {
  it('renders default state', () => {
    const { container } = render(<Input aria-label="test" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders focus state', () => {
    const { container, getByRole } = render(<Input aria-label="test" />);
    fireEvent.focus(getByRole('textbox'));
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders error state', () => {
    const { container } = render(<Input aria-label="test" aria-invalid="true" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('handles aria-invalid="false" as non-error', () => {
    const { container } = render(
      <Input aria-label="test" aria-invalid="false" />
    );
    expect(container.firstChild).not.toHaveClass(
      'border-[hsl(var(--destructive)/0.6)]'
    );
  });

  it('renders disabled state', () => {
    const { container } = render(<Input aria-label="test" disabled />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('adds padding when children are present', () => {
    const { getByRole } = render(
      <Input aria-label="test">
        <span />
      </Input>
    );
    expect(getByRole('textbox')).toHaveClass('pr-7');
  });

  it('adds padding when hasEndSlot is true', () => {
    const { getByRole } = render(
      <Input aria-label="test" hasEndSlot />
    );
    expect(getByRole('textbox')).toHaveClass('pr-7');
  });

  it('has smaller padding by default', () => {
    const { getByRole } = render(<Input aria-label="test" />);
    expect(getByRole('textbox')).not.toHaveClass('pr-7');
  });

  it('has no outline when focused', () => {
    const { getByRole } = render(<Input aria-label="outline" />);
    const input = getByRole('textbox');
    fireEvent.focus(input);
    const style = getComputedStyle(input);
    expect(style.outlineStyle === 'none' || style.outlineStyle === '').toBe(true);
    expect(style.outlineWidth === '0px' || style.outlineWidth === '').toBe(true);
  });

  it('defaults name to generated id', () => {
    const { getByRole } = render(<Input aria-label="name" />);
    const input = getByRole('textbox') as HTMLInputElement;
    expect(input.name).toBe(input.id);
  });

  it('uses slugified label when custom id provided', () => {
    const { getByRole } = render(
      <Input id="email" aria-label="Email Address" />
    );
    const input = getByRole('textbox') as HTMLInputElement;
    expect(input.id).toBe('email');
    expect(input.name).toBe(slugify('Email Address'));
  });

  it('applies rounded-full on pill tone', () => {
    const { container, getByRole } = render(<Input aria-label="pill" tone="pill" />);
    const wrapper = container.firstChild as HTMLElement;
    const input = getByRole('textbox');
    expect(wrapper).toHaveClass('rounded-full');
    expect(input).not.toHaveClass('rounded-full');
  });
});
