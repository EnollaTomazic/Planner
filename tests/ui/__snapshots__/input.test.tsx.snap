// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Input > renders default state 1`] = `
<div
  class="flex w-full flex-col gap-[var(--space-1)]"
>
  <div
    class="group/field relative inline-flex min-h-[var(--field-h,var(--control-h-md))] w-full items-center rounded-[var(--radius-2xl)] border border-[hsl(var(--card-hairline)/0.65)] bg-[hsl(var(--bg))] text-foreground shadow-[inset_0_1px_0_hsl(var(--highlight)/0.08),inset_0_-1px_0_hsl(var(--border)/0.32)] transition-[background,box-shadow,filter] duration-200 ease-standard focus-within:outline-none focus-within:ring-2 focus-within:ring-[hsl(var(--ring))] focus-within:ring-offset-0 focus-within:ring-offset-[hsl(var(--bg))] hover:shadow-[inset_0_1px_0_hsl(var(--highlight)/0.12),inset_0_-1px_0_hsl(var(--border)/0.45)] active:brightness-[0.96] data-[disabled=true]:pointer-events-none data-[disabled=true]:border-[hsl(var(--card-hairline)/0.4)] data-[disabled=true]:bg-[hsl(var(--card))] data-[disabled=true]:shadow-[inset_0_1px_0_hsl(var(--highlight)/0.04)] data-[disabled=true]:text-muted-foreground/70 data-[loading=true]:pointer-events-none data-[loading=true]:opacity-[var(--loading)] data-[invalid=true]:border-[hsl(var(--danger)/0.6)] data-[invalid=true]:focus-within:ring-[hsl(var(--danger))] overflow-hidden"
    style="--field-h: var(--control-h-md);"
  >
    <input
      class="peer block h-[var(--field-h,var(--control-h-md))] w-full rounded-[inherit] border-none bg-transparent px-[var(--space-3)] text-ui text-foreground placeholder:text-muted-foreground/70 caret-accent focus:outline-none focus-visible:outline-none disabled:cursor-not-allowed read-only:cursor-default transition-[color,opacity] group-data-[disabled=true]/field:text-muted-foreground/70 group-data-[disabled=true]/field:placeholder:text-muted-foreground/50 group-data-[loading=true]/field:opacity-[var(--loading)]"
      id=":r0:"
      name=":r0:"
      type="text"
    />
  </div>
</div>
`;

exports[`Input > renders disabled state 1`] = `
<div
  class="flex w-full flex-col gap-[var(--space-1)]"
>
  <div
    aria-disabled="true"
    class="group/field relative inline-flex min-h-[var(--field-h,var(--control-h-md))] w-full items-center rounded-[var(--radius-2xl)] border border-[hsl(var(--card-hairline)/0.65)] bg-[hsl(var(--bg))] text-foreground shadow-[inset_0_1px_0_hsl(var(--highlight)/0.08),inset_0_-1px_0_hsl(var(--border)/0.32)] transition-[background,box-shadow,filter] duration-200 ease-standard focus-within:outline-none focus-within:ring-2 focus-within:ring-[hsl(var(--ring))] focus-within:ring-offset-0 focus-within:ring-offset-[hsl(var(--bg))] hover:shadow-[inset_0_1px_0_hsl(var(--highlight)/0.12),inset_0_-1px_0_hsl(var(--border)/0.45)] active:brightness-[0.96] data-[disabled=true]:pointer-events-none data-[disabled=true]:border-[hsl(var(--card-hairline)/0.4)] data-[disabled=true]:bg-[hsl(var(--card))] data-[disabled=true]:shadow-[inset_0_1px_0_hsl(var(--highlight)/0.04)] data-[disabled=true]:text-muted-foreground/70 data-[loading=true]:pointer-events-none data-[loading=true]:opacity-[var(--loading)] data-[invalid=true]:border-[hsl(var(--danger)/0.6)] data-[invalid=true]:focus-within:ring-[hsl(var(--danger))] overflow-hidden"
    data-disabled="true"
    style="--field-h: var(--control-h-md);"
  >
    <input
      class="peer block h-[var(--field-h,var(--control-h-md))] w-full rounded-[inherit] border-none bg-transparent px-[var(--space-3)] text-ui text-foreground placeholder:text-muted-foreground/70 caret-accent focus:outline-none focus-visible:outline-none disabled:cursor-not-allowed read-only:cursor-default transition-[color,opacity] group-data-[disabled=true]/field:text-muted-foreground/70 group-data-[disabled=true]/field:placeholder:text-muted-foreground/50 group-data-[loading=true]/field:opacity-[var(--loading)]"
      disabled=""
      id=":r4:"
      name=":r4:"
      type="text"
    />
  </div>
</div>
`;

exports[`Input > renders error state 1`] = `
<div
  class="flex w-full flex-col gap-[var(--space-1)]"
>
  <div
    class="group/field relative inline-flex min-h-[var(--field-h,var(--control-h-md))] w-full items-center rounded-[var(--radius-2xl)] border border-[hsl(var(--card-hairline)/0.65)] bg-[hsl(var(--bg))] text-foreground shadow-[inset_0_1px_0_hsl(var(--highlight)/0.08),inset_0_-1px_0_hsl(var(--border)/0.32)] transition-[background,box-shadow,filter] duration-200 ease-standard focus-within:outline-none focus-within:ring-2 focus-within:ring-[hsl(var(--ring))] focus-within:ring-offset-0 focus-within:ring-offset-[hsl(var(--bg))] hover:shadow-[inset_0_1px_0_hsl(var(--highlight)/0.12),inset_0_-1px_0_hsl(var(--border)/0.45)] active:brightness-[0.96] data-[disabled=true]:pointer-events-none data-[disabled=true]:border-[hsl(var(--card-hairline)/0.4)] data-[disabled=true]:bg-[hsl(var(--card))] data-[disabled=true]:shadow-[inset_0_1px_0_hsl(var(--highlight)/0.04)] data-[disabled=true]:text-muted-foreground/70 data-[loading=true]:pointer-events-none data-[loading=true]:opacity-[var(--loading)] data-[invalid=true]:border-[hsl(var(--danger)/0.6)] data-[invalid=true]:focus-within:ring-[hsl(var(--danger))] overflow-hidden"
    data-invalid="true"
    style="--field-h: var(--control-h-md);"
  >
    <input
      aria-invalid="true"
      class="peer block h-[var(--field-h,var(--control-h-md))] w-full rounded-[inherit] border-none bg-transparent px-[var(--space-3)] text-ui text-foreground placeholder:text-muted-foreground/70 caret-accent focus:outline-none focus-visible:outline-none disabled:cursor-not-allowed read-only:cursor-default transition-[color,opacity] group-data-[disabled=true]/field:text-muted-foreground/70 group-data-[disabled=true]/field:placeholder:text-muted-foreground/50 group-data-[loading=true]/field:opacity-[var(--loading)]"
      id=":r2:"
      name=":r2:"
      type="text"
    />
  </div>
</div>
`;

exports[`Input > renders focus state 1`] = `
<div
  class="flex w-full flex-col gap-[var(--space-1)]"
>
  <div
    class="group/field relative inline-flex min-h-[var(--field-h,var(--control-h-md))] w-full items-center rounded-[var(--radius-2xl)] border border-[hsl(var(--card-hairline)/0.65)] bg-[hsl(var(--bg))] text-foreground shadow-[inset_0_1px_0_hsl(var(--highlight)/0.08),inset_0_-1px_0_hsl(var(--border)/0.32)] transition-[background,box-shadow,filter] duration-200 ease-standard focus-within:outline-none focus-within:ring-2 focus-within:ring-[hsl(var(--ring))] focus-within:ring-offset-0 focus-within:ring-offset-[hsl(var(--bg))] hover:shadow-[inset_0_1px_0_hsl(var(--highlight)/0.12),inset_0_-1px_0_hsl(var(--border)/0.45)] active:brightness-[0.96] data-[disabled=true]:pointer-events-none data-[disabled=true]:border-[hsl(var(--card-hairline)/0.4)] data-[disabled=true]:bg-[hsl(var(--card))] data-[disabled=true]:shadow-[inset_0_1px_0_hsl(var(--highlight)/0.04)] data-[disabled=true]:text-muted-foreground/70 data-[loading=true]:pointer-events-none data-[loading=true]:opacity-[var(--loading)] data-[invalid=true]:border-[hsl(var(--danger)/0.6)] data-[invalid=true]:focus-within:ring-[hsl(var(--danger))] overflow-hidden"
    style="--field-h: var(--control-h-md);"
  >
    <input
      class="peer block h-[var(--field-h,var(--control-h-md))] w-full rounded-[inherit] border-none bg-transparent px-[var(--space-3)] text-ui text-foreground placeholder:text-muted-foreground/70 caret-accent focus:outline-none focus-visible:outline-none disabled:cursor-not-allowed read-only:cursor-default transition-[color,opacity] group-data-[disabled=true]/field:text-muted-foreground/70 group-data-[disabled=true]/field:placeholder:text-muted-foreground/50 group-data-[loading=true]/field:opacity-[var(--loading)]"
      id=":r1:"
      name=":r1:"
      type="text"
    />
  </div>
</div>
`;
